app/scripts/modules/google/src/serverGroup/configure/wizard/loadBalancingPolicy/loadBalancingPolicySelector.component.html

```diff
@@ -1,19 +1,56 @@
 <ng-form name="loadBalancingPolicySubForm">
   <collapsible-section heading="Port Name Mapping" expanded="true" subsection="true">
     <div class="form-group">
-      <div class="col-md-5 sm-label-right">
-        <span>http</span><help-field key="gce.serverGroup.loadBalancingPolicy.listeningPort"></help-field>
+      <div class="form-group row" ng-repeat="namedPort in $ctrl.command.loadBalancingPolicy.namedPorts">
+        <div class="col-md-12">
+          <div class="col-md-2 sm-label-right">
+            <b>Name</b>
+            <help-field key="gce.serverGroup.loadBalancingPolicy.portName"></help-field>
+          </div>
+          <div class="col-md-3">
+            <ui-select ng-if="!namedPort.name || !namedPort.name.length"
+                       ng-model="namedPort.name"
+                       class="form-control input-sm">
+              <ui-select-match>{{$item.name}}</ui-select-match>
+              <ui-select-choices repeat="portName in $ctrl.getPortNames()">
+                <span ng-bind-html="portName"></span>
+              </ui-select-choices>
+            </ui-select>
+
+            <input ng-if="namedPort.name.length"
```

Comment:
Why does this need to become a read-only input field after you pick an option? Can't you just pick again from the select field?